warehouse/query-core/src/main/java/datawave/query/discovery/DiscoveryLogic.java

```diff
@@ -379,7 +384,12 @@ public Iterator<DiscoveredThing> apply(Entry<Key,Value> from) {
                 }
                 ArrayList<DiscoveredThing> thangs = Lists.newArrayListWithCapacity(aw.get().length);
                 for (Writable w : aw.get()) {
-                    thangs.add((DiscoveredThing) w);
+                    // Check to see if the field is currently indexed, if it's not, we should NOT be adding it to 'thangs'
+                    if (indexedFields.contains(((DiscoveredThing) w).getField())) {
```

Comment:
opportunity to rename existing variable to something more adult sounding like "items"?

Reply:
I protest, we are not adults here